ClosedXML/Excel/XLWorkbook_Load.cs

```diff
@@ -1324,11 +1324,26 @@ private static void ParseReference(string item, out string sheetName, out string
             }
             else
             {
-                sheetName = sections[0].Replace("\'", "");
+                sheetName = UnescapeName(sections[0]);
                 sheetArea = sections[1];
             }
         }
 
+        private static string UnescapeName(string worksheetName)
+        {
+            return worksheetName
+                .Trim('\'')
+                .Replace("''", "'");
+        }
+
+        private static string EscapeName(string worksheetName)
```

Comment:
I think you should consolidate this function into `StringExtensions.WrapSheetNameInQuotesIfRequired` and rename it to `EscapeSheetName`.

Reply:
If you do that, it should fix the failing tests.